src/extensions/default/ESLint/domain.js

```diff
@@ -0,0 +1,132 @@
+/*eslint no-process-env:0*/
+/*global require, exports*/
+
+(function () {
+  'use strict';
+
+  var fs = require('fs');
+  var CLIEngine = require('eslint').CLIEngine;
+  var cli = new CLIEngine();
+  var currentProjectRoot = null;
+  var domainName = 'brackets-eslint';
+  var domainManager = null;
+  var noop = function () {};
+  var globalPackagesAvailable = false;
+
+  function _setProjectRoot(projectRoot) {
+    var opts = { useEslintrc: true };
+    var configPath;
+    var rulesDirPath;
+    var ignorePath;
+
+    if (projectRoot) {
```

Comment:
This method could be simplified a bit by using [fs.accessSync](https://nodejs.org/api/fs.html#fs_fs_accesssync_path_mode):

``` js
    if (projectRoot) {
        configPath = projectRoot + '.eslintrc';
        fs.accessSync(configPath, function (err) {
            if (err || !fs.statSync(configPath).isFile()) {
                opts.rules = require('eslint/conf/eslint.json').rules;
            }
        });

        rulesDirPath = projectRoot + '.eslintrules';
        fs.accessSync(rulesDirPath, function (err) {
            if (!err && fs.statSync(rulesDirPath).isDirectory()) {
                opts.rulePaths = [rulesDirPath];
            }
        });

        ignorePath = projectRoot + '.eslintignore';
        fs.accessSync(ignorePath, function (err) {
            if (!err && fs.statSync(ignorePath).isFile()) {
                opts.ignore = true;
                opts.ignorePath = ignorePath;
            }
        });
    }
```

This way there's no need for `noop` either. The `fs.accessSync` can be passed an optional mode value too if one just wants to check if the file is readable

```
fs.accessSync("foo", fs.F_OK, function (err) {/*...*/});
```